### DISCUSSION TITLE
issue: Fix is not consistent with errors. isValid Triggers revalidation

### DISCUSSION THREAD

**Discussion by pochka15** [react-hook-form/react-hook-form#9534] [Q&A] [answered]
### Version Number

7.35.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-basic-forked-b5gkw6?file=/src/index.js

### Steps to reproduce

Go to [Sandbox](https://codesandbox.io/s/react-hook-form-basic-forked-b5gkw6?file=/src/index.js)

Experiment 1: 
1. See the message on mount in console: "Why do I have validation on mount?"

Experiment 2:
1. Just remove this line `formState: { isValid, errors }` and remove the disabled property in the submit input
2. See no message in the console: "Why do I have validation on mount?"

### Expected behaviour

Validation is not triggered until I press the submit.

The problem is that I just want to say: disable submit button when the form is invalid. And if there is some invalid field on mount then it becomes impossible to figure out what field contains errors since **errors** object is empty. At first glance it looks very surprising.

Why validation is triggered depending on whether I use `isValid` field or not? So it's enough to trigger validation on start just by logging `isValid`, that's surprising since by default form is validated on submit (if I'm not mistaken)

So to solve my problem i need to use `const isValid = isEmpty(errors)` instead of just using isValid. Could you please clarify why we have this inconsistency?

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
Errors: {} 
Why do I have validation on mount?
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
Because you are subscribed to `isVlaid` `formState`, and in order for react hook form to report the validity of your form, we will have to run the entire form validation.

- `isValid` is dervied for the validty of the form
- `errros` is derived by user action combined with validation strategy `mode`

**Comment by pochka15** [comment]
Thanks for the answer. Don't you think it makes sense to clarify the behaviour In the documentation? E.x it'x given an example in the doc page ([Link](https://react-hook-form.com/api/useform/formstate)). Here a button that listens to `isValid`:

```jsx
// ✅ read all formState values to subscribe to changes
const { isDirty, isValid } = formState;
return <button disabled={!isDirty || !isValid} />; // <--- Here we have isValid
```

I suppose it makes sense to say in the docs that it's better to use `isEmpty(errors)` instead of `isValid` when we have a submit button. Because a submit button is a visible element to user and according to your answer `errros` is derived by user action.

**Reply by bluebill1049** [reply]
> I suppose it makes sense to say in the docs that it's better to use isEmpty(errors) instead of isValid when we have a submit button. 

`isValid` is enough for such usage, you can render an error message as the user changes the input, and they do not conflict with each other.
